completion_aggregator/signals.py

```diff
@@ -40,10 +100,7 @@ def completion_update_handler(signal, sender, instance, created, raw, using, upd
             update_aggregators.delay(
                 user=instance.user,
                 course_key=instance.course_key,
-                block_keys={instance.block_key},
+                block_keys=[instance.block_key],
```

Comment:
Still getting that JSON EncodeError, but it's because of none of these arguments are JSON serialisable: `user`, `course_key`, or the elements in `block_keys`.

I believe you need to pass the `username` and string keys through to the celery task, because they can be serialised, and then fetch the relevant objects from those strings when you need them to create the `AggregationUpdater`.

Reply:
Fixed.